diff --git a/init/main.c b/init/main.c
index 298989b0d4e8..8ab05b881a76 100644
--- a/init/main.c
+++ b/init/main.c
@@ -866,6 +866,11 @@ asmlinkage __visible void __init __no_sanitize_address start_kernel(void)
        pr_notice("%s", linux_banner);
        early_security_init();
        setup_arch(&command_line);
+
+       pr_notice("Origin Kernel command line: %s\n", saved_command_line);
+       pr_notice("modify cmdline for p&w r619ac: %s\n", command_line);
+       command_line = "ubi.mtd=rootfs root=mtd:ubi_rootfs rootfstype=squashfs rootwait root=/dev/ubiblock0_1 rootfstype=squashfs";
+       
        setup_boot_config(command_line);
        setup_command_line(command_line);
        setup_nr_cpu_ids();